### ISSUE TITLE
Crash on Proton-GE 8.7 and above [Linux]

### ISSUE THREAD

**Issue by josh0x10c** [Allen-Wrench/JumpSelector#2]
This is the only plugin that crashes immediately to desktop on initialization. I am not sure exactly what causes it as Space Engineers is not very verbose in it's logging however I have tested this by itself and it is the culprit. I'm not sure which logs would be needed exactly to debug this issue and I'm willing to provide whatever may be needed.

Also I am aware that this game isn't Linux native and that can cause issues, however I would like to bring attention to this issue anyways because more and more people have been leaving Windows behind and deciding to game on Linux nowadays as it becomes more feasible as a gaming platform than ever before. This is the only plugin in all my testing that causes any issues whatsoever.

I have attached a screenshot of me using wine debugger while the game is launching. The left side is Proton-GE 8.6 (last known working version) and the right side is Proton-GE 8.7. I'm not sure if it is how the plugin hooks into the UI that is causing the issue with the later updates to Proton-GE but I suspect so.

Here is a compare with the problematic update of Proton-GE 8.6 to 8.7 on GitHub:
[https://github.com/GloriousEggroll/proton-ge-custom/compare/GE-Proton8-6...GE-Proton8-7](https://github.com/GloriousEggroll/proton-ge-custom/compare/GE-Proton8-6...GE-Proton8-7)

![Screenshot_20230811_195202](https://github.com/Allen-Wrench/JumpSelector/assets/141601785/e17e9a61-e030-46b4-9196-5646c71c18b2)

**Comment by Allen-Wrench** [comment]
On windows the log file that would show why it crashed would be in the users AppData/Roaming/SpaceEngineers folder (not sure what the Proton equivalent folder would be).

Look for a file named similarly to "SpaceEngineers_20230811_180016757.log", it should have a stack trace at the end telling what caused it to crash (hopefully).

**Comment by josh0x10c** [comment]
This is all I get on crash, it never gives a stacktrace and seems to crash on game launch immediately.
https://pastebin.com/qvpsZTHi
I know it's on Linux which makes diagnosing this more challenging so I do appreciate any time or thought you have given this. I think there must have been a change in how Proton works that affects this specific plugin somehow in the way it hooks into the UI but I don't know for sure and haven't been able to get the game to output any useful logs anywhere unfortunately.

**Comment by Allen-Wrench** [comment]
Yeah without a stack trace or some other error pointing to what is causing it to not work, there's not really much I can do. I'm not doing anything special with the UI, just calling the original methods that the rest of the game uses. 

Have you tried disabling all of the other plugins or starting the game without using the plugin loader game launcher? As long as the harmony .dll is in the same folder as the plugin, you can just use the launch option '-plugin C:\location\of\plugin.dll' and see if that works.

**Comment by arcvvolf** [comment]
This crash is also happening in Proton-GE 8-25

The game loads into saves fine but if you add the jump select option to the toolbar and then push the hotkey for jump selector the game crashes to the desktop.

The proton log has a reference to `EXCEPTION handling: System.MissingMethodException: Method not found: void Sandbox.Graphics.GUI.MyGuiControls.Add(Sandbox.Graphics.GUI.MyGuiControlBase)`

The SpaceEngineers log has the following error:
`2023-11-27 13:20:59.737 - Thread:   1 ->  Exception occurred: System.MissingMethodException: Method not found: void Sandbox.Graphics.GUI.MyGuiControls.Add(Sandbox.Graphics.GUI.MyGuiControlBase)
  at JumpSelector.Plugin.JumpSelectorGui..ctor (Sandbox.Game.Entities.MyJumpDrive block) [0x00167] in <1e7d1b15f2404930b905151ba19d89c0>:0 
  at JumpSelector.Plugin.JumpSelectorPatch.ShowJumpSelector (Sandbox.Game.Entities.MyJumpDrive block) [0x00020] in <1e7d1b15f2404930b905151ba19d89c0>:0 
  at Sandbox.Game.Gui.MyTerminalAction`1[TBlock].<set_Action>b__20_0 (TBlock block, VRage.Collections.ListReader`1[T] parameters) [0x00000] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.Game.Gui.MyTerminalAction`1[TBlock].Apply (Sandbox.Game.Entities.Cube.MyTerminalBlock block, VRage.Collections.ListReader`1[T] parameters) [0x00023] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.Game.Screens.Helpers.MyToolbarItemTerminalBlock.Activate () [0x00024] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.Game.Screens.Helpers.MyToolbar.ActivateItemAtIndex (System.Int32 index, System.Boolean checkIfWantsToBeActivated) [0x0007b] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.Game.Screens.Helpers.MyToolbar.ActivateItemAtSlot (System.Int32 slot, System.Boolean checkIfWantsToBeActivated, System.Boolean playActivationSound, System.Boolean userActivated) [0x00024] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.Game.Screens.Helpers.MyToolbarComponent.HandleInput () [0x000c0] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.Game.World.MySession.HandleInput () [0x0001a] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.MySandboxGame.Update () [0x006cb] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.Engine.Platform.Game.UpdateInternal () [0x00083] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.Engine.Platform.Game.RunSingleFrame () [0x0000d] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.Engine.Platform.FixedLoop+<>c__DisplayClass11_0.<Run>b__0 () [0x0003c] in <16be29f54c1742bd8b0d97920285afd9>:0 
  at Sandbox.Engine.Platform.GenericLoop.Run (Sandbox.Engine.Platform.GenericLoop+VoidAction tickCallback) [0x00009] in <16be29f54c1742bd8b0d97920285afd9>:0 
  at Sandbox.Engine.Platform.FixedLoop.Run (Sandbox.Engine.Platform.GenericLoop+VoidAction tickCallback) [0x00014] in <16be29f54c1742bd8b0d97920285afd9>:0 
  at Sandbox.Engine.Platform.Game.RunLoop () [0x00053] in <e2180f3666b24280a78f023298df657b>:0 
  at Sandbox.MySandboxGame.Run (System.Boolean customRenderLoop, System.Action disposeSplashScreen) [0x000dd] in <e2180f3666b24280a78f023298df657b>:0 
  at SpaceEngineers.MyProgram.Main (System.String[] args) [0x00383] in <dea25e824fe54ba09780dd58ba1abc73>:0 
  at avaness.SpaceEngineersLauncher.Program.StartSpaceEngineers (System.String[] args) [0x00000] in <3df4883e7d184b019d048eabd8995cb0>:0 
  at avaness.SpaceEngineersLauncher.Program.Main (System.String[] args) [0x00099] in <3df4883e7d184b019d048eabd8995cb0>:0 
2023-11-27 13:20:59.737 - Thread:   1 ->  Showing message
2023-11-27 13:20:59.738 - Thread:   1 ->  Exception while reporting crash.
2023-11-27 13:20:59.739 - Thread:   1 ->  System.NotImplementedException: The method or operation is not implemented.
  at VRage.Platform.Windows.MyCrashReporting.WriteMiniDump (System.String path, VRage.MyMiniDump+Options options, System.IntPtr exceptionPointers) [0x0001f] in <a9d7ba7ff1074d47b87eadbb272d196c>:0 
  at VRage.MyMiniDump.CollectCrashDump (System.IntPtr exceptionPointers) [0x00020] in <51d7560edc864435bda92ffffc45a435>:0 
  at Sandbox.MyInitializer.OnCrash (System.String logPath, System.String gameName, System.String minimumRequirementsPage, System.Boolean requiresDX11, System.Exception exception, System.IntPtr exceptionPointers, System.Boolean oom) [0x0008c] in <e2180f3666b24280a78f023298df657b>:0 
2023-11-27 13:20:59.739 - Thread:   1 ->  Log Closed`


I hope this helps shed some light on the problem.
